demos/coreMQTT/mqtt_demo_basic_tls.c

```diff
@@ -0,0 +1,1012 @@
+        if( xDemoStatus == pdPASS )
+        {
+            LogInfo( ( "Unsubscribe from the MQTT topic %s.\r\n", mqttexampleTOPIC ) );
+            xDemoStatus = prvMQTTUnsubscribeFromTopic( &xMQTTContext );
+        }
+
+        if( xDemoStatus == pdPASS )
+        {
+            /* Process incoming UNSUBACK packet from the broker. */
+            xMQTTStatus = MQTT_ProcessLoop( &xMQTTContext, mqttexamplePROCESS_LOOP_TIMEOUT_MS );
+
+            if( xMQTTStatus != MQTTSuccess )
+            {
+                LogError( ( "MQTT_ProcessLoop() call failed to receive UNSUBACK response."
+                            " Error=%s.",
+                            MQTT_Status_strerror( xMQTTStatus ) ) );
+                xDemoStatus = pdFAIL;
+            }
+        }
+
+        /*************************** Disconnect. ******************************/
+
+        /* Whether the demo operations preceeding are successful or not, we
+         * always disconnect an open connection to free up system resources. */
+        if( xIsConnectionEstablished == pdTRUE )
+        {
+            /* Send an MQTT DISCONNECT packet over the already-connected TLS over TCP connection.
+             * There is no corresponding response expected from the broker. After sending the
+             * disconnect request, the client must close the network connection. */
+            LogInfo( ( "Disconnecting the MQTT connection with %s.\r\n", democonfigMQTT_BROKER_ENDPOINT ) );
+            xMQTTStatus = MQTT_Disconnect( &xMQTTContext );
+
+            if( xMQTTStatus != MQTTSuccess )
+            {
+                LogError( ( "MQTT_Disconnect() failed with status %s.",
+                            MQTT_Status_strerror( xMQTTStatus ) ) );
+                xDemoStatus = pdFAIL;
+            }
+
+            /* Close the network connection.  */
+            xNetworkStatus = SecureSocketsTransport_Disconnect( &xNetworkContext );
+
+            if( xNetworkStatus != TRANSPORT_SOCKET_STATUS_SUCCESS )
+            {
+                xDemoStatus = pdFAIL;
+                LogError( ( "SecureSocketsTransport_Disconnect() failed to close "
+                            "the network connection with status code %d.",
+                            ( int ) xNetworkStatus ) );
+            }
+            else
+            {
+                xIsConnectionEstablished = pdFALSE;
+            }
+        }
+
+        if( xDemoStatus == pdPASS )
+        {
+            /* Reset SUBACK status for each topic filter after completion of
+             * subscription request cycle. */
+            for( ulTopicCount = 0; ulTopicCount < mqttexampleTOPIC_COUNT; ulTopicCount++ )
+            {
+                xTopicFilterContext[ ulTopicCount ].xSubAckStatus = MQTTSubAckFailure;
+            }
+
+            LogInfo( ( "RunCoreMqttBasicTLSDemo() completed an iteration successfully. "
+                       "Total free heap is %u.\r\n",
+                       xPortGetFreeHeapSize() ) );
```

Comment:
Cypress is giving false positives in the custom build. It doesn't have this function defined, could you please update like this recent change: https://github.com/aws/amazon-freertos/commit/65b8537f3c2ee45b6088d2d4418f4b1d89d14085